spine-c/src/spine/SkeletonBinary.c

```diff
@@ -0,0 +1,1009 @@
+	attachment->verticesCount = kv_size(weights);
+	attachment->vertices = kv_array(weights);
+
+	kv_trim(int, bones);
+	attachment->bonesCount = kv_size(bones);
+	attachment->bones = kv_array(bones);
+}
+
+spAttachment* spSkeletonBinary_readAttachment(spSkeletonBinary* self, _dataInput* input,
+		spSkin* skin, int slotIndex, const char* attachmentName, int/*bool*/ nonessential) {
+	int i;
+	spAttachmentType type;
+	const char* name = readString(input);
+	if (!name) MALLOC_STR(name, attachmentName);
+
+	type = (spAttachmentType)readByte(input);
+
+	switch (type) {
+		case SP_ATTACHMENT_REGION: {
+			const char* path = readString(input);
+			spAttachment* attachment;
+			spRegionAttachment* region;
+			if (!path) MALLOC_STR(path, name);
+			attachment = spAttachmentLoader_createAttachment(
+					self->attachmentLoader, skin, type, name, path);
+			region = SUB_CAST(spRegionAttachment, attachment);
+			region->path = path;
+			region->rotation = readFloat(input);
+			region->x = readFloat(input) * self->scale;
+			region->y = readFloat(input) * self->scale;
+			region->scaleX = readFloat(input);
+			region->scaleY = readFloat(input);
+			region->width = readFloat(input) * self->scale;
+			region->height = readFloat(input) * self->scale;
+			readColor(input, &region->r, &region->g, &region->b, &region->a);
+			spRegionAttachment_updateOffset(region);
+			spAttachmentLoader_configureAttachment(self->attachmentLoader, attachment);
+			return attachment;
+		}
+		case SP_ATTACHMENT_BOUNDING_BOX: {
+			int vertexCount = readVarint(input, 1);
+			spAttachment* attachment = spAttachmentLoader_createAttachment(
+					self->attachmentLoader, skin, type, name, 0);
+			_readVertices(self, input, SUB_CAST(spVertexAttachment, attachment), vertexCount);
+			if (nonessential) readInt(input); /* Skip color. */
+			spAttachmentLoader_configureAttachment(self->attachmentLoader, attachment);
+			return attachment;
+		}
+		case SP_ATTACHMENT_MESH: {
+			int vertexCount;
+			spAttachment* attachment;
+			spMeshAttachment* mesh;
+			const char* path = readString(input);
+			if (!path) MALLOC_STR(path, name);
+			attachment = spAttachmentLoader_createAttachment(self->attachmentLoader, skin, type, name, path);
+			mesh = SUB_CAST(spMeshAttachment, attachment);
+			mesh->path = path;
+			readColor(input, &mesh->r, &mesh->g, &mesh->b, &mesh->a);
+			vertexCount = readVarint(input, 1);
+			mesh->regionUVs = _readFloatArray(input, vertexCount << 1, 1);
+			mesh->triangles = (unsigned short*)_readShortArray(input, &mesh->trianglesCount);
+			_readVertices(self, input, SUPER(mesh), vertexCount);
+			spMeshAttachment_updateUVs(mesh);
+			mesh->hullLength = readVarint(input, 1) << 1;
+			if (nonessential) {
+				mesh->edges = (int*)_readShortArray(input, &mesh->edgesCount);
+				mesh->width = readFloat(input) * self->scale;
+				mesh->height = readFloat(input) * self->scale;
+			} else {
+				mesh->edges = 0;
+				mesh->width = 0;
+				mesh->height = 0;
+			}
+			return attachment;
```

Comment:
Gold catch! Thanks! It seems like there is already a PR that fixes this problem https://github.com/EsotericSoftware/spine-runtimes/pull/688
Btw we're still on the way to actually use it in our cocos2d game, does it work for you with this fix without other problems?